ncm-afsclt/src/main/perl/afsclt.pm

```diff
@@ -2,997 +2,314 @@
-      my $nextname = $next->getName();
-      my $nextvalue = $next->getValue();
-      $options .= " " if $options;
-      $options .= "-$nextname $nextvalue";
-      $have_config = 1;
-    }
-  }
-  # "settime" option
-  if($config->elementExists("$mypath/settime")) {
-    $node = $config->getElement("$mypath/settime");
-    $have_config = 1;
-    my $settime = $node->getValue();
-    if(! $settime) {  # caution, not set = assume NTP
-      $options .= " " if $options;
-      $options .= "-nosettime"
-    }
-  }
-  if($have_config) {
-    $ret = NCM::Check::lines($sysconfig_afs,
-			     linere => '^#.*$compname-options|OPTIONS=.*',
-			     goodre => '^#.*$compname-options|OPTIONS=\s*"'.quotemeta($options).'"',
-			     good   => "# changed by $compname-options\nOPTIONS=\"".$options."\"",
-			     keep   => 'first',
-			     add    => 'last',
-			     backup => '.old_options');
-  } else {
-    $self->info("no OPTIONS configuration for $sysconfig_afs, skipping");
-  }
-
-  # cache size is configured in a separate function
-
-  if($config->elementExists("$mypath/verbose")) {
-    $node = $config->getElement("$mypath/verbose");
-    my $verbose = $node->getValue() ? '-verbose' : '';
-    $ret &= NCM::Check::lines($sysconfig_afs,
-			     linere => '^#.*$compname-verbose|VERBOSE=.*',
-			     goodre => '^#.*$compname-verbose|VERBOSE=\s*'.quotemeta($verbose),
-			     good   => "# changed by $compname-verbose\nVERBOSE=".$verbose,
-			     keep   => 'first',
-			     add    => 'last',
-			     backup => '.old_verbose');
-  }
-
-  if($config->elementExists("$mypath/enabled")) {
-    $node = $config->getElement("$mypath/enabled");
-    my $enabled = $node->getValue(); #syntax needs to be on/off, not yes/no
-    if ($enabled =~ /^off$|^no$/i) {
-      $enabled = 'off';
-    } else {
-      $enabled = 'on'; #default
+    my $proc = CAF::Process->new( [ "fs", "getcacheparms" ], log => $self, keeps_state => 1 );
+    if ( !( $? >> 8 ) && $proc->output() =~ /AFS using \d+ of the cache's available (\d+) (\w+) byte blocks/ ) {
+        if ( $2 ne "1K" ) {
+            $self->error("Cannot handle $2 (non-1K) AFS cache block sizes");
+            return 0;
+        }
+        $run_cache = $1;
+    }
+    else {
+        $self->info("Cannot determine current AFS cache size, changing only config file");
+    }
+
+    my $afs_cacheinfo_fh = CAF::FileEditor->new( $afs_cacheinfo, log => $self );
+    if ( "$afs_cacheinfo_fh" =~ m;^([^:]+):([^:]+):(\d+)$; ) {
+        $file_afsmount   = $1;
+        $file_cachemount = $2;
+        $file_cache      = $3;
+    }
+    else {
+        $self->error("Cannot parse stored AFS cache mount or size from $afs_cacheinfo");
+        return 0;
+    }
+    $afs_cacheinfo_fh->close();
+
+    # sanity check - don't allow cachesize bigger than 95% of a partition size
+    $proc = CAF::Process->new( [ "df", "-k", $file_cachemount ], log => $self, keeps_state => 1 );
+    for ( split ( "\n", $proc->output() ) ) {
```

Comment:
better not use `$_` in perl, it really does not behave as you expect. this better becomes
```perl
foreach my $line split ( "\n", $proc->output() ) ) {
   if ($line =~ m{...}) {
...
```
(`$_` is good for oneliners like `grep {$_ eq 'something} @array`, but for eveything else, keep it readable and use a variable)